### PR TITLE
Fixed PR358

### PR DESCRIPTION
**Pull request by mark0978**
This cleans up the objections to PR358 which was adding a better error message for json parse errors

### CODE REVIEW

**Comment by mark0978** [comment]
I have no idea how to fix the flake8 invocation error.

File: cookiecutter/generate.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
 from jinja2.exceptions import TemplateSyntaxError
 from binaryornot.check import is_binary
 
-from .exceptions import NonTemplatedInputDirException
+from .exceptions import NonTemplatedInputDirException, ContextDecodingException
+from collections import OrderedDict
```
Comment (michaeljoseph) [inline cookiecutter/generate.py]: Duplicate import.

**Comment by michaeljoseph** [comment]
@mark0978: these linting errors are resonably self-explanatory?
```
cookiecutter/generate.py:24:1: F811 redefinition of unused 'OrderedDict' from line 11
cookiecutter/generate.py:48:80: E501 line too long (99 > 79 characters)
cookiecutter/generate.py:52:80: E501 line too long (100 > 79 characters)
```
#390 fixes the new E731 pep8 check.

**Comment by pydanny** [comment]
Reset the builds but am still getting flake8 issues. @mark0978, when you update I'll run them again.

**Comment by mark0978** [comment]
The flake 8 error I was seeing was an invocation error.  Not even sure
where to start trouble shooting that one

File: cookiecutter/generate.py [hunk 2]
```diff
@@ -41,7 +42,16 @@ def generate_context(context_file='cookiecutter.json', default_context=None,
...
 
     file_handle = open(context_file)
-    obj = json.load(file_handle, object_pairs_hook=OrderedDict)
+    try:
+        obj = json.load(file_handle, object_pairs_hook=OrderedDict)
+    except ValueError as e:
+        # JSON decoding error.  Let's throw a new exception that is more friendly for the developer
+        # or user.
+        full_fpath = os.path.abspath(context_file)
+        json_exc_message = str(e)
```
Comment (pydanny) [inline cookiecutter/generate.py]: Shouldn't this use ``traceback.format_exc``?

:bell: @michaeljoseph

**Comment by pydanny** [comment]
@mark0978 To fix the flake8 issue, all you should have to do is rebase from master, then commit to this PR. Travis will pick up the changes and it will run. :checkered_flag:

File: cookiecutter/generate.py (hunk 2 above)
Comment (mark0978) [inline cookiecutter/generate.py]: I don't think so.  We aren't trying to debug the application here, we are trying to debug the json file.  A traceback would run counter to that purpose.

**Comment by pydanny** [comment]
I'm going to dig into this one. @mark0978, however the code gets merged in you'll get credit in the HISTORY.

**Comment by mark0978** [comment]
Sorry I haven't had time to get on a windows machine to see what's up.
Been a bit busy here lately

**Comment by blackrosezy** [comment]
It would be nice if the error message can be output in json format. Something like this:
```
$ cookiecutter test_dir
{
    "error" : "Decoding error details: 'Expecting property name: line 4 column 1 (char 53)'"
}
```
Just my two cents :)

**Pull request merged**